nautobot/ipam/tests/test_filters.py

```diff
@@ -579,70 +555,88 @@ def setUpTestData(cls):
         )
 
     def test_search(self):
-        search_terms = {
-            # string searches
-            "ipaddress-a": 2,
-            "foo": 0,
-            # network searches
-            "": 10,
-            "10": 5,
-            "10.": 5,
-            "10.0": 5,
-            "10.0.0.4": 1,
-            "10.0.0.4/24": 1,
-            "11": 0,
-            "11.": 0,
-            "11.0": 0,
-            "10.10.10.0/24": 0,
-            "2001": 5,
-            "2001:": 5,
-            "2001::": 5,
-            "2001:db8:": 5,
-            "2001:db8::": 5,
-            "2001:db8::/64": 5,
-            "2001:db8::2": 1,
-            "2001:db8:0:2": 0,
-            "fe80": 0,
-            "fe80::": 0,
-            "foo.bar": 0,
-        }
-
-        for term, cnt in search_terms.items():
-            params = {"q": term}
-            self.assertEqual(self.filterset(params, self.queryset).qs.count(), cnt)
+        ipv4_address = self.queryset.ip_family(4).exclude(dns_name="").exclude(dns_name__isnull=True).first()
+        ipv6_address = self.queryset.ip_family(6).exclude(dns_name="").exclude(dns_name__isnull=True).first()
+        ipv4_octets = ipv4_address.host.split(".")
+        ipv6_hextets = ipv6_address.host.split(":")
+
+        search_terms = [
+            str(ipv4_address.address),  # ipv4 address with mask: 10.0.0.1/24
+            ipv4_address.host,  # ipv4 address without mask: 10.0.0.1
+            str(ipv6_address.address),  # ipv6 address with mask: 2001:db8::1/64
+            ipv6_address.host,  # ipv6 address without mask: 2001:db8::1
+            ipv4_address.dns_name,
+            ipv4_octets[0],  # 10
+            f"{ipv4_octets[0]}.",  # 10.
+            f"{ipv4_octets[0]}.{ipv4_octets[1]}",  # 10.0
+            f"{ipv4_octets[0]}.{ipv4_octets[1]}.",  # 10.0.
+            ipv6_hextets[0],  # 2001
+            f"{ipv6_hextets[0]}:",  # 2001:
+            f"{ipv6_hextets[0]}::",  # 2001::
+            "no match",
+        ]
+
+        for term in search_terms:
+            with self.subTest(term):
+                params = {"q": term}
+                self.assertQuerysetEqual(self.filterset(params, self.queryset).qs, self.queryset.string_search(term))
```

Comment:
Wish we could make this one a bit more robust - as written it'll pass if `string_search()` returns no matches for every term, right?

I wonder if we might want to create a custom test method like `assertQuerysetEqualAndNotEmpty()` to use in all of the filter tests (not just those in this PR)?

It's less of a concern for tests where we're using standard queryset filters (`vrf__isnull=False` and the like) as it'd be pretty weird for those to not work as expected, but for all of these ones involving custom filters (`string_search`, `net_*`) I'm concerned.

Reply:
I was thinking the same when adding this line to the test but ended up going this route anyway because there are already tests to test that `string_search` is working properly in [test_querysets.py](https://github.com/nautobot/nautobot/blob/0333de62a936d15761e8725747d7d4045a31d7df/nautobot/ipam/tests/test_querysets.py#L76). Since this is testing that the filterset is working properly, I was satisfied that the combination of these tests and the next test that's making sure a blank query returns all objects would be sufficient. I wrote this test using an existing object's properties to ensure that even with randomized data the filter should always return at least 1 object but an `assertQuerysetEqualAndNotEmpty()` would help to enforce that.

Reply:
I filed #2633 to track the `assertQuerysetEqualAndNotEmpty` idea.